libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKModule.kt

```diff
@@ -147,12 +233,23 @@ class BreezSDKModule(reactContext: ReactApplicationContext) : ReactContextBaseJa
 
     @ReactMethod
     fun start(promise: Promise) {
-        try {
-            getBreezServices().start()
-            promise.resolve(readableMapOf("status" to "ok"))
-        } catch (e: SdkException) {
-            e.printStackTrace()
-            promise.reject(TAG, e.message ?: "Error calling start", e)
+        val requestId = getRequestId(promise)
+
+        executor.execute {
+            val message = Message.obtain(null, MSG_RESPONSE, requestId, 0)
+            val data = Bundle()
+
+            try {
+                getBreezServices().start()
+                data.putString("data", serialize(readableMapOf("status" to "ok")))
+            } catch (e: SdkException) {
+                e.printStackTrace()
+                message.what = MSG_ERROR
+                data.putString("error", e.message ?: "Error calling start")
+            }
+
+            message.data = data
+            responseHandler.sendMessage(message)
         }
```

Comment:
Am I on the right track @roeierez? The executor would run this block to start the Breez SDK in a thread from the thread pool.